packages/flutter_tools/test/commands.shard/hermetic/doctor_test.dart

```diff
@@ -93,11 +98,14 @@ void main() {
       expect(validator.plistFile, '/path/to/app/Contents/Info.plist');
       expect(validator.pluginsPath, pluginsDirectory.path);
     }, overrides: <Type, Generator>{
-      Platform: () => FakePlatform()
-        ..environment = <String, String>{'HOME': '/foo/bar'},
+      Platform: () => macPlatform,
       PlistParser: () => mockPlistParser,
       FileSystem: () => fileSystem,
       ProcessManager: () => mockProcessManager,
+      FileSystemUtils: () => FileSystemUtils(
```

Comment:
Seemed to be calling real filesystem, but I'm not sure how

Reply:
Shouldn't the default `FileSystemUtils` use the file system and platform from the context (and thus the overridden test mocks)?

And if that's not happening, would removing contructor injection `FileSystemUtils` fix this? This worries me that you can override the ProcessManager + FileSystem but still end up using the real filelsystem.